ambari-server/src/main/java/org/apache/ambari/server/controller/utilities/RemovableIdentities.java

```diff
@@ -133,13 +131,16 @@ private RemovableIdentities(List<KerberosIdentityDescriptor> candidateIdentities
   }
 
   /**
-   * Remove all identities which are not used by other services or components
+   * Remove all identities which are related to the specified set of components and not used by
+   * other services or components
    */
   public void remove(KerberosHelper kerberosHelper) throws AmbariException, KerberosOperationException {
-    Set<String> identitiesToRemove = skipUsed().stream().map(KerberosIdentityDescriptor::getPath).collect(toSet());
-    if (!identitiesToRemove.isEmpty()) {
-      kerberosHelper.deleteIdentities(cluster, components, identitiesToRemove);
-    }
+    // TODO: Fix the implementation identifying specific identities in the event we need to  pinpoint
```

Comment:
It is a TODO that we should think about for later. I am not sure this is really needed since we Ambari does not mess with specific identities. It messes with groups of identities related to hosts, services, or components.